redisinsight/ui/src/components/base/display/progress-bar/progress-bar-loader.styles.ts

```diff
@@ -0,0 +1,90 @@
+import { theme, useTheme } from '@redis-ui/styles'
+import { ReactNode } from 'react'
+import styled, { css, keyframes } from 'styled-components'
+
+export type EuiColorNames =
+  | 'inherit'
+  | 'default'
+  | 'primary'
+  | 'danger'
+  | 'warning'
+  | 'success'
+
+interface LoaderBarProps {
+  color?: string
+}
+
+export type ColorType = LoaderBarProps['color'] | EuiColorNames | (string & {})
+type ThemeColors = typeof theme.semantic.color
+
+export const getBarBackgroundColor = (
+  color: ColorType,
+  themeColors: ThemeColors,
+) => {
+  if (!color) {
+    return themeColors.background.primary300
+  }
+
+  const barBackgroundColors: Map<ColorType, string> = new Map([
+    ['inherit', 'inherit'],
+    ['default', themeColors.background.primary300],
+    ['primary', themeColors.background.primary300],
+    ['danger', themeColors.background.danger600],
+    ['warning', themeColors.background.attention600],
+    ['success', themeColors.background.success600],
+  ])
```

Comment:
This seems way too complicated. Is there any reason to use map?
```suggestion
  const barBackgroundColors: Record<ColorType, string> = {
    inherit: 'inherit',
    default: themeColors.background.primary300,
    primary: themeColors.background.primary300,
    danger: themeColors.background.danger600,
    warning: themeColors.background.attention600,
    success: themeColors.background.success600,
  }
```

Reply:
As we spoke offline, I plan to get this logic outside in a util, and pass this map as a parameter. This map would be different for link, text, backgrounds etc. Why is it complicated, it is pretty much the same as the switch case...

Reply:
turns out i have misunderstood - you were talking about why not Record but Map, I was thinking why not switch case, since we are using it in other places for the same functionality. changed to Record. thx!